index.js

```diff
@@ -13,60 +25,117 @@ const QUESTIONS = [{
-  // body should have email and password
+  const { email, password } = req.body
 
-  // Check if the user with the given email exists in the USERS array
-  // Also ensure that the password is the same
+  // Find the user with the given email in the USERS array
+  const user = USERS.find(user => user.email === email)
 
+  // Check if a user was found
+  if (!user) {
+    // If the user doesn't exist or the password is incorrect, return 401 Unauthorized status code
+    res.status(401).send('Invalid email!')
+    return
+  }
 
-  // If the password is the same, return back 200 status code to the client
-  // Also send back a token (any random string will do for now)
-  // If the password is not the same, return back 401 status code to the client
+  // Check if the password is correct
+  if (user.password !== password) {
+    // If the user doesn't exist or the password is incorrect, return 401 Unauthorized status code
+    res.status(401).send('Wrong password!')
+    return
+  }
 
+  // If the user exists and the password is correct, generate a random token string for authentication
+  const token = Math.random().toString(36).substring(7)
 
-  res.send('Hello World from route 2!')
-})
+  // Store the token in the user object for future authentication
+  user.token = token
 
-app.get('/questions', function(req, res) {
+  // Return back 200 status code to the client along with the token
+  res.status(200).json({ token, isAdmin: user.isAdmin })
+});
 
+
+app.get('/questions', function(req, res) {
   //return the user all the questions in the QUESTIONS array
-  res.send("Hello World from route 3!")
+  res.json(QUESTIONS);
+})
+
+// leaving as hard todos
+
+// Create a route that lets an admin add a new problem
+// ensure that only admins can do that.
+
+app.post("/questions", function(req, res) {
+  // Add logic to decode body
+  const { title, description, testCases } = req.body
+
+  // Check if user is admin, then only user can submit a problem
+  if(USERS[1].isAdmin){ // since we are not storing anything in DB, just checking manually out of 2 users
+    // Store the submission in the SUBMISSIONS array
+    QUESTIONS.push({
+      title: title,
+      description: description,
+      testCases: testCases
+    })
+
+    // Return a response to the client
+    res.status(200).send("Problem Submitted!")
+  }
+  // Else return not an admin
+  res.status(400).send("Not an Admin!")
+  
 })
 
 app.get("/submissions", function(req, res) {
-   // return the users submissions for this problem
-  res.send("Hello World from route 4!")
+  // return the users submissions for this problem
+  res.json(SUBMISSIONS);
 });
 
 
 app.post("/submissions", function(req, res) {
-   // let the user submit a problem, randomly accept or reject the solution
-   // Store the submission in the SUBMISSION array above
-  res.send("Hello World from route 4!")
-});
+  // Add logic to decode body
+  const { problem, solution } = req.body
 
-// leaving as hard todos
```

Comment:
where is the hard todo like you mentioned in the PR title?

Reply:
it's here LOC 97 : https://github.com/hkirat/full-stack-assignment/pull/224/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R95

Reply:
I left a comment there, it's not implemented correctly